### ISSUE TITLE
Users reports of 'This extension has been corrupted' on install...?!

### ISSUE THREAD

**Issue by june07** [june07/NiM#34]
I've not been able to duplicate on my end.  Maybe someone has an idea about what this might be and would like to lend a hand...

**Comment by melanholly** [comment]
First of all this extension is real time server. 
Second I got `This extension has been corrupted` every time I click on the toolbar icon. 
* My operating system is `Linux Mint 18 Sarah`
* Node version is `v8.9.0`
* NIM version is `0.14.2`


If you need more information or some logs I will provide them.

**Comment by mckaydavis** [comment]
(tl;dr temporary workaround at bottom)

I get this error consistently on my Linux (`Ubuntu 17.10`) Chrome with no other extensions installed.

details from `chrome://version`:  
**Google Chrome:** `61.0.3163.79 (Official Build) (64-bit)`
**Revision:** `2bd57c7d956c47bc51eff132dd114136ec0a6db7-refs/branch-heads/3163@{#1092}`

Steps to reproduce:
* Install extension `v0.14.2` [from the chrome web store](https://chrome.google.com/webstore/detail/nodejs-v8-inspector-manag/gnhhdgbaldcilmgcpfddgdbkhjohddkj)
* The NiM icon appears in the extension toolbar
* Click on the icon in the toolbar
* The icon disappears (presumably because something happened to make Chrome mark the extension 'corrupted')
* Navigate to `chrome://extensions`
* The error message reads: `This extension may have been corrupted` in red
* Click `Repair` -> Popup -> `Repair extension` and the extension is re-downloaded and re-enabled
* Attempts to debug via the `inspect views: background.html` fail.  When the toolbar icon is clicked and the extension is "corrupted" the debug window is immediately closed.
* After the extension is repaired: *Clicking `Options` and the `options.html` opens fine*

Which leads to the temporary workaround:

**If the user never clicks on the extension toolbar icon, NiM functions correctly and can be configured via the `Options` link in the `chrome://extensions` interface.**

**Comment by mckaydavis** [comment]
This is nigh impossible to debug as installing it from an unpacked local dir works fine.
My steps:
* From the command line:
```
cp -r ~/.config/google-chrome/Default/Extensions/gnhhdgbaldcilmgcpfddgdbkhjohddkj/0.14.2_0 /tmp/nim
rm -rf /tmp/nim/_metadata # chrome won't load extension w/ _metadata dir
```
* Navigate to `chrome://extensions`
* Enable the `Developer mode` checkbox
* Make sure no existing NiM extension is installed
* Click the `Load unpacked extension...` button
* In the `Select the extension directory.` dialog open the `/tmp/nim` dir

Everything works fine now...(i.e. clicking the extension toolbar icon does not cause chrome to mark the extension 'corrupted')

*I have no idea...*

**Comment by june07** [comment]
@mckaydavis THANKS for your work on this!

**Comment by ga-tb** [comment]
I see the same problem.

My operating system is `Ubuntu 17.10`
Node version is `v8.8.1`
NIM version is `0.14.3`
Chrome version is `62.0.3202.94 (Official Build) (64-bit)`

The workaround from @mckaydavis wasn't enough for me because my app uses a custom port and url, and I couldn't see a way to set these from the options page.

I made an attempt to debug it... Alas; the problem disappeared when I attempted to observe it!

**Steps**

* Enable `Developer Mode` for chrome extensions
* Disable the extension
* Click `Load Unpacked Extension`
* Navigate to the extension directory, for me that's `~/.config/google-chrome/Default/Extensions/gnhhdgbaldcilmgcpfddgdbkhjohddkj/0.14.3_0`
* Open and Enable

Now everything works as expected... :confused: 

There is one warning that appeared on the extension page during developer mode. Not sure if it's relevant:

```
There were warnings when trying to install this extension:
_metadata is a reserved directory that will not be allowed at the time of Chrome Web Store upload.
```

**Comment by june07** [comment]
https://bugs.chromium.org/p/chromium/issues/detail?id=785372&q=component%3APlatform>Extensions%20corrupted&colspec=ID%20Pri%20M%20Stars%20ReleaseBlock%20Component%20Status%20Owner%20Summary%20OS%20Modified

**Comment by june07** [comment]
For a temporary workaround, installing the CRX https://github.com/june07/NiM/releases file directly should work.  Can anyone confirm...

**Comment by mckaydavis** [comment]
@june07 :+1: confirmed the CRX works.

I just downloaded [NiM-0.14.5.crx](https://github.com/june07/NiM/releases/download/0.14.5/NiM-0.14.5.crx) and dragged the CRX onto the Chrome window to install.

To further verify, I uninstalled NiM and installed it from the [chrome web store](https://chrome.google.com/webstore/detail/nodejs-v8-inspector-manag/gnhhdgbaldcilmgcpfddgdbkhjohddkj).  It still shows the same 'This extension has been corrupted' behavior. :cry:

**Comment by june07** [comment]
Yeah, I spent more time that I would have liked trying to track down the root cause!?  For now I hope the majority of users experiencing the problem will find this workaround!

@mckaydavis THANK YOU for your continued support and feedback on this!  Helped a ton having someone to work with on it.

**Comment by hellsinglord22** [comment]
Just in case anyone trying to drag and drop, do this within the `chrome://extensions`

**Comment by jnv** [comment]
I wonder, it looks like only Linux is affected? I used to get this error on Ubuntu 17.04 some time ago, now on macOS with Canary NiM works fine from WebStore.

**Comment by june07** [comment]
@hellsinglord22 Thanks.  Will update install page to reflect that important tidbit!

**Comment by june07** [comment]
@jnv No, I'm seeing reports from every OS including macOS and Windows, and I had it happen to me in ChromeOS.

**Comment by Kamshak** [comment]
Happens for me. Ubuntu, Drag Drop does the same thing as normal store install, crashes when you press the icon. Can use options from the chrome://extensions tab

**Comment by kjarnet** [comment]
Same as Kamshak :(

**Comment by heisian** [comment]
Same issue here - works in Chromium, but in Chrome, the icon disappears as soon as I click it.

**Comment by june07** [comment]
Any ideas?

**Comment by heisian** [comment]
Not sure, Chromium is unstable and crashes on occasion when a new window/tab is opened automatically by NIM. It also starts to accumulate higher and higher CPU usage to the point of slowing everything down. This running on an 8700K. Definitely not NIM's fault - but I would like to have used it in the stable release of Chrome - which I can't.

I'm appalled at how debugging in Node is treated as an afterthought - whereas in any other mature language debugging is a core part of the developer experience. But that's a topic for another thread...

**Comment by gtarsia** [comment]
Can you guys retry? I think this is fixed on chrome extension v2.0.2.
